### PR TITLE
Example how to use environment variables

### PR DESCRIPTION
**Pull request by JohanWork**
Small example on how to set a environment variable, pass it in to the container and the printing it. 

This likes to issue: https://github.com/kubeflow/website/issues/759#issuecomment-584378574

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/website/1697)

### CODE REVIEW

**Comment by JohanWork** [comment]
@NikeNano can you take a look ?

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
++++
+title = "Using enviroment varibales in piplines"
+description = "How set and use enviroment variables in kubeflow piplines"
```
Comment (NikeNano) [inline content/docs/pipelines/sdk/enviroment_variables.md]: ”How to set ......”

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
++++
+
+This page describes how to create a enviroment varaibale for Kubeflow Pipelines 
+and how to combine enviroment variable into a pipeline.
```
Comment (NikeNano) [inline content/docs/pipelines/sdk/enviroment_variables.md]: Maybe you could change the last part. It sounds like the env variables create a pipeline.

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+This page describes how to create a enviroment varaibale for Kubeflow Pipelines 
+and how to combine enviroment variable into a pipeline. 
+
+## Before you start
+
+### Initial setup
+Before starting with this tutorial the following things have to be in place: 
+- A GCP account.
+- [Kubeflow set up on GKE](https://www.kubeflow.org/docs/gke/deploy/deploy-cli/)
```
Comment (NikeNano) [inline content/docs/pipelines/sdk/enviroment_variables.md]: Do it need to be on gcp?

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+Then we transform the function to a container using the 
+`kfp.components.func_to_container_op(func)`.  
+```python
+    logg_env_function_op = comp.func_to_container_op(logg_env_function, base_image='tensorflow/tensorflow:1.11.0-py3')
+```
+
+Then the functions is put into a pipline and  it is here that we mount the 
+enviroment variable to the container. This code is the same no matter if your
+using python lightweight components or prebuild container.
```
Comment (NikeNano) [inline content/docs/pipelines/sdk/enviroment_variables.md]: Good and important comment!

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    name='Logging',
+    description='A pipline showing how to use enviroment variables'
+    )
+    def enviroment_pipline():
+
+        #Returns a dsl.ContainerOp class instance. 
+        container_op = logg_env_function_op().add_env_variable(V1EnvVar(name='Test_env', value='env_variable')) 
+```
+
+If more varaiblaes is wanted you would simply add one more .add_env_varaible()
```
Comment (NikeNano) [inline content/docs/pipelines/sdk/enviroment_variables.md]: Maybe change this sentence a bit for better grammar.

File: content/docs/pipelines/sdk/enviroment_variables.md (hunk 1 above)
Comment (JohanWork) [reply content/docs/pipelines/sdk/enviroment_variables.md]: Will change!

File: content/docs/pipelines/sdk/enviroment_variables.md (hunk 2 above)
Comment (JohanWork) [reply content/docs/pipelines/sdk/enviroment_variables.md]: Will change, see what you mean!

File: content/docs/pipelines/sdk/enviroment_variables.md (hunk 3 above)
Comment (JohanWork) [reply content/docs/pipelines/sdk/enviroment_variables.md]: True it dose not!

File: content/docs/pipelines/sdk/enviroment_variables.md (hunk 4 above)
Comment (JohanWork) [reply content/docs/pipelines/sdk/enviroment_variables.md]: Thnx!

**Comment by joeliedtke** [comment]
Hi @JohanWork, thank you for writing this doc. It looks like it will be a helpful addition. Sorry for the delay, I'll be working on the review for this PR tomorrow.

**Comment by JohanWork** [comment]
No problems, @joeliedtke looking forward for you feedback/inputs!

**Comment by joeliedtke** [comment]
This page can be previewed at:
https://deploy-preview-1697--competent-brattain-de2d6d.netlify.com/docs/pipelines/sdk/

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+description = "How to set and use enviroment variables in kubeflow piplines"
++++
+
+This page describes how to create a enviroment varaibale for Kubeflow Pipelines 
+and how they work in a pipline. 
+
+## Before you start
+
+### Initial setup
+Before starting with this tutorial the following things have to be in place:
```
Comment (joeliedtke) [inline content/docs/pipelines/sdk/enviroment_variables.md]: Suggested revision:

Set up your environment(Shorter wording, use general KF installation page):

* [Install Kubeflow](/docs/started/getting-started/)
* [Install the Kubeflow Pipelines SDK](/docs/pipelines/sdk/install-sdk/)

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+### Initial setup
+Before starting with this tutorial the following things have to be in place: 
+- [Kubeflow set up on kubernetes](https://www.kubeflow.org/docs/gke/deploy/deploy-cli/)
+- Jupyter nooteboks set up on you kubeflow deployment
+
+
+## Using enviroment variables 
+
+Below is a example python lightweight component where we read a enviroment
```
Comment (joeliedtke) [inline content/docs/pipelines/sdk/enviroment_variables.md]: It is a good practice in technical docs not to use "we". I can become confusing for the reader to determine who is acting. More information is available in the [style guide](https://www.kubeflow.org/docs/about/style-guide/#address-the-audience-directly)

Suggested revision:

In this example, you pass an environment variable to a lightweight Python component, which writes the variable's value to the log.

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+variable and the logging it out as a string. 
+
+We are going to use jupyter ligthweight componetes you can find more about them
```
Comment (joeliedtke) [inline content/docs/pipelines/sdk/enviroment_variables.md]: Suggested revision(avoiding we):

Learn more about [building lightweight pipeline components](/docs/pipelines/sdk/lightweight-python-components/).

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+Below is a example python lightweight component where we read a enviroment
+variable and the logging it out as a string. 
+
+We are going to use jupyter ligthweight componetes you can find more about them 
+[here](https://www.kubeflow.org/docs/pipelines/sdk/lightweight-python-components/).
+
+To build a component, define a stand-alone Python function and then call 
+`kfp.components.func_to_container_op(func)` to convert the function to a 
+component that can be used in a pipeline. Below is one small function logging
```
Comment (joeliedtke) [inline content/docs/pipelines/sdk/enviroment_variables.md]: Replace the following:

Below is one small function logging out the enviroment variable.

With the following:

The following function gets an environment variable and writes it to the log.

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    def logg_env_function():
+        import os
+        import logging
+        logging.basicConfig(level=logging.INFO)
+        env_variable = os.getenv('Test_env')
+        #logging out the enviroment variable
+        logging.info('The enviroment variable is: {}'.format(env_variable))
+```
+
+Then we transform the function to a container using the
```
Comment (joeliedtke) [inline content/docs/pipelines/sdk/enviroment_variables.md]: Suggested revision(Avoid we):

Transform the function into a component using [`kfp.components.func_to_container_op(func)`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.components.html#kfp.components.func_to_container_op).

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+        logging.info('The enviroment variable is: {}'.format(env_variable))
+```
+
+Then we transform the function to a container using the 
+`kfp.components.func_to_container_op(func)`.  
+```python
+    logg_env_function_op = comp.func_to_container_op(logg_env_function, base_image='tensorflow/tensorflow:1.11.0-py3')
+```
+
+Then the functions is put into a pipline and  it is here that we mount the
```
Comment (joeliedtke) [inline content/docs/pipelines/sdk/enviroment_variables.md]: Suggested revision(shorter, avoids we):

Add this component to a pipeline. Use [`add_env_variable()`](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.dsl.html#kfp.dsl.ContainerOp.container) to pass an environment variable into the component.

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+    name='Logging',
+    description='A pipline showing how to use enviroment variables'
+    )
+    def enviroment_pipline():
+
+        #Returns a dsl.ContainerOp class instance. 
+        container_op = logg_env_function_op().add_env_variable(V1EnvVar(name='Test_env', value='env_variable')) 
+```
+
+To use more then one env variables you simply add on more .add_env_varaible()
```
Comment (joeliedtke) [inline content/docs/pipelines/sdk/enviroment_variables.md]: Suggested revision:

To pass more environment variables into a component, add more instances of `add_env_variable()`. Use the following command to run this pipeline using the Kubeflow Pipelines SDK.

File: content/docs/pipelines/sdk/enviroment_variables.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+`kfp.components.func_to_container_op(func)` to convert the function to a 
+component that can be used in a pipeline. Below is one small function logging 
+out the enviroment variable.
+
+```python
+    def logg_env_function():
+        import os
+        import logging
+        logging.basicConfig(level=logging.INFO)
+        env_variable = os.getenv('Test_env')
```
Comment (joeliedtke) [inline content/docs/pipelines/sdk/enviroment_variables.md]: Consider renaming the environment variable to something like 'example_env'

**Comment by JohanWork** [comment]
@joeliedtke I believe you had a lot of good inputs have update now.

**Comment by joeliedtke** [comment]
/lgtm
/approve

Thanks for working on this doc!
